### DISCUSSION TITLE
Zigbee RGB Automatic color change mode

### DISCUSSION THREAD

**Discussion by BassT23** [arendst/Tasmota#16004] [Q&A] [answered]
Hi

is it possible to make a automatic color change mode for RGB Lights within Zigbee?

**Comment by sfromis** [comment, marked as answer]
1) Depends on the light, if it includes automation
2) You can create automations (in Tasmota or whatever) to do repeated commands via Zigbee.

**Reply by BassT23** [reply]
I dont know if light had this included.
As Coordinator I use an CC2531 With Tasmota flashed. 
The Lights are Cskyzk E27 Bulb.
I set the Lightmode to 5, but Alexa didn't show any automatic, so I cant see anything in console.

I see a "ColorMode:0" but, don't know if this is for that. Also don't know how to change that for testing.

For now I help me out with an ioBroker Blockly script, that change the Hue color every 3 seconds +10, but it would be better if this can be implemented in Tasmota, for better network performance. So there don't need to send an http request every 3 seconds, ...

**Reply by sfromis** [reply]
You are already doing it "wrong" if you use http requests, it is better to use MQTT.

And using [Rules](https://tasmota.github.io/docs/Rules/), it is certainly possible to create an automation in Tasmota to do repeated commands to change colors. If you use ESP32, a more friendly programming setup is the Berry scripting language.

While I know nothing about that bulb, my general expectation would be that Zigbee bulbs focus on basic features, unless it was advertised with light show effects.

**Reply by BassT23** [reply]
I use an d1 mini 8266 - Tasmota is v10.0.0 for now. Scripting not enabled during compiling for now.
Could make an new one with v12.0.2, if it work better then.
the request goes over mqtt, for example: http://192.168.21.190/cm?cmnd=ZbSend {"Group":"200","Send":{"Hue":100}}
How to make this as an rule? Any examples for cooking ;)

**Reply by sfromis** [reply]
The page I linked includes a "cookbook" with many examples.

And your example is http, not the better option of MQTT.

What I meant by "Berry scripting" is ESP32-only. Yes, there is a very different scripting feature for ESP8266 to replace rules, be prepared for a steeper learning curve.

In general, I'd suggest getting to newest version of Tasmota when wanting to do something, matches current docs, may have new useful features, and will have fixes for old issues.

**Reply by BassT23** [reply]
ok, tomorrow I compile a new 12.0.2 version with scripting enabled.
are there any examples for the script language to do - if trigger that -> toggle hue from 0 to 250?
how can I trigger that? - from ioBroker

**Reply by sfromis** [reply]
Well, I meant "steeper learning curve" quite literally, takes some time to grok it (I've chosen to abstain). Of course, the docs page for the [Scripting Language](https://tasmota.github.io/docs/Scripting-Language/) has examples too. The general concept for anything done via Zigbee is that you need to construct ZbSend commands, like you already know, any command can also be issued via the scripting language.

For simpler stuff (?), you may find the Rules feature more accessible, even though it still looks weird to start with.

**Reply by BassT23** [reply]
Thx for the suggestions.
A lot of work and learn.
Will start tomorrow ;)
